OSX 10.11.2 El Capitan compile error (cddlib)

IRIS fails to compile in my computer running OSX 10.11.2 El Capitan because cddlib fails to compile.

I get the errors:

```
/bin/sh ../libtool --tag=CC   --mode=link gcc  -g -O2 -L/usr/local/lib  -o fourier_gmp fourier.o ../lib-src-gmp/libcddgmp.la
libtool: link: gcc -g -O2 -o .libs/fourier_gmp fourier.o  -L/usr/local/lib ../lib-src-gmp/.libs/libcddgmp.dylib
Undefined symbols for architecture x86_64:
  "___gmpq_clear", referenced from:
      _main in fourier.o
  "___gmpq_init", referenced from:
      _main in fourier.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [fourier_gmp] Error 1
make: *** [all-recursive] Error 1
```

That originate when trying to compile the folder `src-gmp`.

The problem is that it should be linking gmp, but it doesn't.
That is, from the `src-gmp` folder the make file is trying to do:

```
bin/sh ../libtool --tag=CC   --mode=link gcc  -g -O2 -L/usr/local/lib  -o fourier_gmp fourier.o ../lib-src-gmp/libcddgmp.la
```
which fails. However, adding -lgmp:

```
bin/sh ../libtool --tag=CC   --mode=link gcc  -g -O2 -L/usr/local/lib  -o fourier_gmp fourier.o ../lib-src-gmp/libcddgmp.la -lgmp
```
works fine. This happens for all the .o compontents of `src-gmp`.

Logically this also fails when trying to install Drake WITH_IRIS=ON.

A quick crappy fix consists on changing line 258 of `Makefile` in `cdd_project/src-gmp/Makefile` from:

```
LDADD = ../lib-src-gmp/libcddgmp.la
```

to

```
LDADD = ../lib-src-gmp/libcddgmp.la -lgmp
```

Then Drake builds properly.

The really weird thing is that I'm able to install cddlib using homebrew from source without issues. The formula is [here](https://github.com/Homebrew/homebrew-science/blob/master/cddlib.rb) but it doesn't seem to be doing anything at all about this. The homebrew formula points to an older version of cddlib (0.94g) while IRIS downloads (0.94h). However, I've tried to compile both versions standalone and I get the same errors unless adding the -lgmp flag.


Has anyone else seen this using OSX El Capitan?

Thanks!

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,10 +31,11 @@ if (IRIS_WITH_CDD)
 	set(CDD_PROJECT_SRC_DIR ${CMAKE_CURRENT_BINARY_DIR}/cdd_project-prefix/src/cdd_project)
 	ExternalProject_Add(cdd_project
 		URL ftp://ftp.ifor.math.ethz.ch/pub/fukuda/cdd/cddlib-094h.tar.gz
-	    CONFIGURE_COMMAND ${CDD_PROJECT_SRC_DIR}/configure --prefix=${CMAKE_INSTALL_PREFIX}
-	    BUILD_COMMAND make
-	    BUILD_IN_SOURCE 1
-	    )
+		PATCH_COMMAND patch -p1 < ${CMAKE_CURRENT_SOURCE_DIR}/cdd.patch
+    CONFIGURE_COMMAND ${CDD_PROJECT_SRC_DIR}/configure --prefix=${CMAKE_INSTALL_PREFIX}
+    BUILD_COMMAND make
+    BUILD_IN_SOURCE 1
+    )
 	set(iris_dependencies ${iris_dependencies} cdd_project)
 endif()
 

--- a/cdd.patch
+++ b/cdd.patch
@@ -1,0 +1,20 @@
+diff -p --exclude=.DS_Store -r cddlib-094h 2/src-gmp/Makefile.in cddlib-094h/src-gmp/Makefile.in
+*** cddlib-094h 2/src-gmp/Makefile.in	2012-03-23 04:51:54.000000000 -0400
+--- cddlib-094h/src-gmp/Makefile.in	2016-02-18 13:37:52.000000000 -0500
+*************** testlp1_gmp_SOURCES = testlp1.c
+*** 255,261 ****
+  testlp2_gmp_SOURCES = testlp2.c
+  testlp3_gmp_SOURCES = testlp3.c
+  # cddmathlink_SOURCES                = cddmathlink.c cddmlio.h cddmlio.c
+! LDADD = ../lib-src-gmp/libcddgmp.la
+  AM_LDFLAGS = -L$(gmplibdir)
+  INCLUDES = -I../lib-src-gmp -I$(gmpincludedir)
+  AM_CPPFLAGS = -DGMPRATIONAL
+--- 255,261 ----
+  testlp2_gmp_SOURCES = testlp2.c
+  testlp3_gmp_SOURCES = testlp3.c
+  # cddmathlink_SOURCES                = cddmathlink.c cddmlio.h cddmlio.c
+! LDADD = ../lib-src-gmp/libcddgmp.la -lgmp
+  AM_LDFLAGS = -L$(gmplibdir)
+  INCLUDES = -I../lib-src-gmp -I$(gmpincludedir)
+  AM_CPPFLAGS = -DGMPRATIONAL

